### PR TITLE
Fix video play/pause state after DND

### LINKED ISSUE #2618: Can't play the video sometimes (no play button)
## Bug Description

From Sam:
If I drag a video and set as bg, the video autoplays. Then I deselect the video (video stops playing) then select it again but there is no play button so I'm not able to ever play the video again.

First thought: Might be connected to isTransforming and dropTargets that don’t set null.

## Additional Context

- Plugin Version:
- Operating System:
- Browser:

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### LINKED ISSUE #4024: Setting BG Video - Play/Pause State Logic
PR4012 - Win10/Chrome

- added a video to canvas
- clicked Set As Background
- Video was set as background, but with "pause" button appearing (meaning video should be playing)
- Video wasn't playing

- Note: If I click the "pause" button and THEN "play", then it will begin to play

Video - https://drive.google.com/file/d/15Xzw7miM1BuvArf3EWFuQyrsSB_Rn5R5/view

### PR DESCRIPTION
**Pull request by merapi**
## Relevant Technical Choices

Just pushing `null` was not enough, we first want to reset the styles, then communicate that drop is finished with `null`.

## Testing Instructions

Drop video from the library to canvas background, then deselect/move the mouse out and check if you can pause/play the video (see the controls).

---

Fixes #2618, Fixes https://github.com/google/web-stories-wp/issues/4024

### CODE REVIEW

File: assets/src/edit-story/elements/video/controls.js [hunk 1]
```diff
@@ -239,7 +241,7 @@ function VideoControls({
           timeout={100}
         >
           <ButtonWrapper
-            title={buttonTitle}
+            aria-label={buttonLabel}
```
Comment (swissspidy) [inline assets/src/edit-story/elements/video/controls.js:244]: 👍

File: assets/src/edit-story/elements/video/karma/autoplay.karma.js [hunk 2]
```diff
@@ -68,5 +73,26 @@ describe('Autoplay video', () => {
...
+    const pauseButton = fixture.screen.getByRole('button', {
+      name: 'Click to pause',
+    });
+    await fixture.events.click(pauseButton);
+    expect(backgroundElVideo.paused).toBe(true);
+    const playButton = fixture.screen.getByRole('button', {
+      name: 'Click to play',
+    });
+    await fixture.events.click(playButton);
+    expect(backgroundElVideo.paused).toBe(false);
```
Comment (barklund) [inline assets/src/edit-story/elements/video/karma/autoplay.karma.js]: Could we rewrite this (entire test) to use [the "fake DOM" container structure](https://github.com/google/web-stories-wp/blob/fix/video-play-edge-cases/assets/src/edit-story/karma/fixture/containers/editor.js#L29)? And add the remaining properties and containers there if missing?

Comment (barklund) [reply assets/src/edit-story/elements/video/karma/autoplay.karma.js]: Also can we wrap it in [a proper CUJ for this](https://github.com/google/web-stories-wp/issues/1587) - probably something akin to `describe('CUJ: Creator can Play/Preview Video: Play and pause and reset video' () => {})` (can't use a comma in the action, thus the rewrite to an extra _"and"_).

**Review by barklund — APPROVED** [review]
Looks good and seems to work smoothly. However, I'd like to see the karma test updated to the new fake DOM container structure (cleaner test and more DRY accessors) and adding CUJ annotations.

File: assets/src/edit-story/elements/video/karma/autoplay.karma.js (hunk 2 above)
Comment (merapi) [reply assets/src/edit-story/elements/video/karma/autoplay.karma.js]: @barklund PTAL at a rewrite, what more would you put into fixture.editor?

CUJ: I think we can cover that in separate PR (general cleanup), that particular CUJ says nothing about autoplay, and its tasks are spread over various files.

File: assets/src/edit-story/components/canvas/layout.js [hunk 3]
```diff
@@ -24,6 +24,10 @@ import { forwardRef, createRef } from 'react';
 /**
  * Internal dependencies
  */
+/**
+ * WordPress dependencies
+ */
```
Comment (swissspidy) [inline assets/src/edit-story/components/canvas/layout.js]: Need to clean up the comments a little bit

File: assets/src/edit-story/components/canvas/layout.js [hunk 4]
```diff
@@ -221,7 +225,12 @@ const PageArea = forwardRef(function PageArea(
   ref
 ) {
   return (
-    <PageAreaFullbleedContainer ref={fullbleedRef} data-testid="fullbleed">
+    <PageAreaFullbleedContainer
+      ref={fullbleedRef}
+      data-testid="fullbleed"
+      aria-label={__('Fullbleed', 'web-stories')}
```
Comment (swissspidy) [inline assets/src/edit-story/components/canvas/layout.js]: "Fullbleed" is not really descriptive. Maybe "Fullbleed area" or something? How does VoiceOver announce it?

File: assets/src/edit-story/components/canvas/layout.js (hunk 3 above)
Comment (merapi) [reply assets/src/edit-story/components/canvas/layout.js]: Done in https://github.com/google/web-stories-wp/pull/3643/commits/331240015ac564463578ff42d828ff2667432cda

File: assets/src/edit-story/elements/video/karma/autoplay.karma.js [hunk 5]
```diff
@@ -39,34 +44,58 @@ describe('Autoplay video', () => {
...
+      .display(backgroundId)
+      .node.querySelector('video');
+    expect(backgroundElVideo.paused).toBe(false);
+
+    // Bug #2618 coverage
+    await fixture.events.mouse.seq(({ moveRel }) => [
+      moveRel(fullbleed, '10%', '10%'),
+    ]);
+    await waitFor(() =>
+      fixture.screen.getByRole('button', { name: 'Click to pause' })
```
Comment (barklund) [inline assets/src/edit-story/elements/video/karma/autoplay.karma.js]: This one should come from the container structure too. I'm thinking from the frame or display element, whichever has the actual button as a child.

File: assets/src/edit-story/components/canvas/layout.js (hunk 4 above)
Comment (merapi) [reply assets/src/edit-story/components/canvas/layout.js]: The same applies to e.g. [Display](https://github.com/google/web-stories-wp/blob/7cf07181c4a39995df3552a4f50b86036c2b43af/assets/src/edit-story/components/canvas/displayLayer.js#L54) or [Frames](https://github.com/google/web-stories-wp/blob/7cf07181c4a39995df3552a4f50b86036c2b43af/assets/src/edit-story/components/canvas/framesLayer.js#L93) @barklund WDYT about naming them? Display layer, Frames layer, Fullbleed area?

File: assets/src/edit-story/elements/video/karma/autoplay.karma.js (hunk 5 above)
Comment (merapi) [reply assets/src/edit-story/elements/video/karma/autoplay.karma.js]: Done in https://github.com/google/web-stories-wp/pull/3643/commits/1a7f95e734f94224d98a0bf762ab4dcb3687ffa8

File: assets/src/edit-story/components/canvas/layout.js (hunk 4 above)
Comment (merapi) [reply assets/src/edit-story/components/canvas/layout.js]: Done in https://github.com/google/web-stories-wp/pull/3643/commits/5e98f9339367d3aa955054fb02904dd47c2e4bae

**Comment by swissspidy** [comment]
@merapi Can you fix the conflicts here & merge this?
